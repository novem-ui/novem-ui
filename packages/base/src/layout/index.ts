import { css } from '@emotion/react'
import { LayoutProps } from '../types'

const layout = ({ display, top, left, right, bottom, position }: LayoutProps) => {
  const style = css`
    ${display && `display: ${display};`}
    ${top && `top: ${top};`}
    ${left && `left: ${left};`}
    ${right && `right: ${right};`}
    ${bottom && `bottom: ${bottom};`}
    ${position && `position: ${position};`}
  `

  return style
}

export default layout
