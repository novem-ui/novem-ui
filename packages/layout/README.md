# @novem-ui/layout

## Usage

### `<Box />`

The component props are composed by [`BorderProps & BackgroundProps & LayoutProps`](../base/src/types.ts) & [SpacingProps](../base/src/spacing/README.md) & [ShadowProps](../base/src/shadows/index.ts)

You can use any color of the palette in the color props by doing <color>.<weight>.
The color can be any from the theme, and the valid weights are `50 | 100 | 200 | 300 | 400 | 500 | 600 | 700 | 800 | 900`

```tsx
import { Box, ThemeProvider } from '@novem-ui/core'
// or
import { Box } from '@novem-ui/layout'
import { ThemeProvider } from '@novem-ui/theme'

function App() {
  return (
    <ThemeProvider>
      <Box backgroundColor="red.300" />
    </ThemeProvider>
  )
}
```

### `<Flex />`

It's the same as Box but includes [`FlexProps`](../base/src/types.ts)

```tsx
import { Flex, ThemeProvider } from '@novem-ui/core'
// or
import { Flex } from '@novem-ui/layout'
import { ThemeProvider } from '@novem-ui/theme'

function App() {
  return (
    <ThemeProvider>
      <Flex align="center" justify="left" />
    </ThemeProvider>
  )
}
```
