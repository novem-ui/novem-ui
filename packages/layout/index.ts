export * from './src/box'
export * from './src/flex'
