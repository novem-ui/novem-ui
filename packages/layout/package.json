{
  "name": "@novem-ui/layout",
  "version": "0.0.2-alpha.0",
  "description": "layout components from novem-ui",
  "keywords": [
    "layout",
    "ui",
    "component",
    "react",
    "orange"
  ],
  "author": "Leonel Vieyra <me@leonelv.com>",
  "homepage": "https://github.com/jdrets/novem-ui/tree/main/packages/layout#readme",
  "license": "ISC",
  "main": "index.js",
  "publishConfig": {
    "access": "public",
    "directory": "../../dist/layout"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/jdrets/novem-ui.git"
  },
  "scripts": {
    "test": "jest --env=jsdom --passWithNoTests",
    "build": "yarn build:esm && yarn build:types && yarn build:package-json",
    "build:esm": "../../node_modules/.bin/babel . --root-mode upward --extensions .ts,.tsx -d ../../dist/layout --source-maps",
    "build:types": "tsc -p ./tsconfig.types.json",
    "build:package-json": "../../scripts/create-package.js"
  },
  "bugs": {
    "url": "https://github.com/jdrets/novem-ui/issues"
  },
  "peerDependencies": {
    "@emotion/react": "^11.1.5",
    "@emotion/styled": "^11.1.5",
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
  },
  "devDependencies": {
    "@novem-ui/theme": "^0.0.2-alpha.0",
    "@types/novem-theme": "^0.0.2-alpha.0"
  },
  "dependencies": {
    "@novem-ui/base": "^0.0.2-alpha.0"
  }
}
