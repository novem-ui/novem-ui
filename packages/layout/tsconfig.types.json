{
  "extends": "../../tsconfig.json",
  "files": [],
  "compilerOptions": {
    "emitDeclarationOnly": true,
    "declaration": true,
    "declarationDir": "../../dist/layout",
    "declarationMap": true,
    "types": ["novem-theme"]
  },
  "include": ["**/*.ts", "**/*.tsx"],
  "exclude": ["node_modules/**", "dist/**", "**/*.spec.ts", "**/*.spec.tsx", "**/*.stories.tsx"]
}
