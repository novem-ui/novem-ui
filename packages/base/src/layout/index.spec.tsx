import React from 'react'

import { render } from '@testing-library/react'
import styled from '@emotion/styled'

import layout from '.'
import type { LayoutProps } from '../types'

const ComponentWithLayoutProps = styled.div<LayoutProps>`
  ${layout}
`

describe('layout', () => {
  test('should create a css fragment with the mapped props (partial)', () => {
    const { container } = render(<ComponentWithLayoutProps top="10px" />)

    expect(container).toMatchSnapshot()
  })

  test('should create a css fragment with the mapped props (including all props)', () => {
    const props: LayoutProps = {
      top: '10px',
      left: '10px',
      right: '10px',
      bottom: '10px',
      position: 'relative',
      display: 'flex',
      width: '10px',
      height: '10px',
      maxWidth: '100px',
      maxHeight: '50px',
      zIndex: 10
    }

    const { container } = render(<ComponentWithLayoutProps {...props} />)

    expect(container).toMatchSnapshot()
  })
})
