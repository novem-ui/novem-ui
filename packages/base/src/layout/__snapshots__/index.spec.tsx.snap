// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout should create a css fragment with the mapped props (including all props) 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  top: 10px;
  left: 10px;
  right: 10px;
  bottom: 10px;
  position: relative;
}

<div>
  <div
    class="emotion-0 emotion-1"
    display="flex"
  />
</div>
`;

exports[`layout should create a css fragment with the mapped props (partial) 1`] = `
.emotion-0 {
  top: 10px;
}

<div>
  <div
    class="emotion-0 emotion-1"
  />
</div>
`;
